akka-docs/rst/java/untyped-actors.rst

```diff
@@ -714,6 +714,73 @@ before stopping the target actor. Simple cleanup tasks can be handled in ``postS
+
+The default phases are defined in a single linear order, but the phases can be ordered as a 
+directed acyclic graph (DAG) by defining the dependencies between the phases.
+The phases are ordered with `topological <https://en.wikipedia.org/wiki/Topological_sorting>`_ sort of the DAG. 
+
+Tasks can be added to a phase with:
+
+.. includecode:: code/docs/actorlambda/ActorDocTest.java#coordinated-shutdown-addTask
+
+The returned ``CompletionStage<Done>`` should be completed when the task is completed. The task name parameter
+is only used for debugging/logging. 
+
+Tasks added to the same phase are executed in parallel without any ordering assumptions. 
+Next phase will not start until all tasks of previous phase have been completed.
+
+If tasks are not completed within a configured timeout (see :ref:`reference.conf <config-akka-actor>`)
+the next phase will be started anyway. It is possible to configure ``recover=off`` for a phase
+to abort the rest of the shutdown process if a task fails or is not completed within the timeout.
+
+Tasks should typically be registered as early as possible after system startup. When running 
+the coordinated shutdown tasks that have been registered will be performed but tasks that are 
+added too late will not be run.
+
+To start the coordinated shutdown process you can invoke ``runAll`` on the ``CoordinatedShutdown``
+extension:
+
+.. includecode:: code/docs/actorlambda/ActorDocTest.java#coordinated-shutdown-run
+
+It's safe to call the ``runAll`` method multiple times. It will only run once.
+
+That also means that the ``ActorSystem`` will be terminated in the last phase. By default, the
+JVM is not forcefully stopped (it will be stopped if all non-daemon threads have been terminated).
+To enable a hard ``System.exit`` as a final action you can configure::
+
+  akka.coordinated-shutdown.exit-jvm = on
+
+When using :ref:`Akka Cluster <cluster_usage_java>` the ``CoordinatedShutdown`` will automatically run
+when the cluster node sees itself as ``Exiting``, i.e. leaving from another node will trigger
+the shutdown process on the leaving node. Tasks for graceful leaving of cluster including graceful 
+shutdown of Cluster Singletons and Cluster Sharding are added automatically when Akka Cluster is used, 
+i.e. running the shutdown process will also trigger the graceful leaving if it's not already in progress. 
+
+By default, the ``CoordinatedShutdown`` will be run when the JVM process exits, e.g.
+via ``kill SIGTERM`` signal (``SIGINT`` ctrl-c doesn't work). This behavior can be disabled with::
+
+  akka.coordinated-shutdown.run-by-jvm-shutdown-hook=off
+
+If you have application specific JVM shutdown hooks it's recommended that you register them via the
+``CoordinatedShutdown`` so that they are running before Akka internal shutdown hooks, e.g. 
+those shutting down Akka Remoting (Artery).
+
+.. includecode:: code/docs/actorlambda/ActorDocTest.java#coordinated-shutdown-jvm-hook
```

Comment:
Though I was tempted to make this a top-level section, WDYT? Since it relates to other things too, not just "actors".